pkg/gateway/api/api_grpc_test.go

```diff
@@ -4073,3 +4073,89 @@ func emptyUserEvaluations(t *testing.T) *featureproto.UserEvaluations {
 		ForceUpdate:        false,
 	}
 }
+
+func TestGrpcListFeatures(t *testing.T) {
+	t.Parallel()
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	patterns := []struct {
+		desc        string
+		ctx         context.Context
+		setup       func(*grpcGatewayService)
+		expected    *gwproto.ListFeaturesResponse
+		expectedErr error
+	}{
+		{
+			desc: "fails: bad role",
+			setup: func(gs *grpcGatewayService) {
+				gs.environmentAPIKeyCache.(*cachev3mock.MockEnvironmentAPIKeyCache).EXPECT().Get(gomock.Any()).Return(
+					&accountproto.EnvironmentAPIKey{
+						Environment: &environmentproto.EnvironmentV2{Id: "ns0"},
+						ApiKey: &accountproto.APIKey{
+							Id:       "id-0",
+							Role:     accountproto.APIKey_SDK_SERVER,
+							Disabled: false,
+						},
+					}, nil)
+			},
+			expected:    nil,
+			expectedErr: ErrBadRole,
```

Comment:
How about checking the missing API key?

Reply:
IMO, that case is one for checkRequest test.
In TestGrpcGetFeature, I need 2 cases; success or failure.